app/models/membership.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+#
+# Model for relationships between Groups and Users.
+# Updates the group status on creation and destruction.
+# Validations:
+#   - Users can only join groups in their draw
+#   - Groups can only be joined when they are open
+#
+# @attr group [Group] The group of the membership.
+# @attr user [User] The user of the membership.
+class Membership < ApplicationRecord
+  belongs_to :group, counter_cache: true
+  belongs_to :user
+
+  validates :group, presence: true
+  validates :user, presence: true
+  validate :matching_draw, if: proc { |m| m.user.present? && m.group.present? }
+
+  validate :group_is_open, if: proc { |m| m.group.present? }, on: :create
+
+  after_create :update_group_status
```

Comment:
I think we should depend on the group status being accurate. A single membership shouldn't need to know how many other members there are in the group. 

Also, I'm not sure how things would get messed up on modification; right now there isn't anything to modify. I think we can move this conversation into how we handle #56, but I don't think it will be too much of a problem.

I added the count vs size validation back into groups, though.